overlord/snapstate/snapstate.go

```diff
@@ -1091,7 +1096,11 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		updated = append(updated, name)
 	}
 
-	if len(updated) > 0 && !globalFlags.NoReRefresh {
+	return updated, tasksets, nil
+}
+
+func finalizeUpdate(st *state.State, tasksets []*state.TaskSet, hasUpdates bool, updated []string, userID int, globalFlags *Flags) []*state.TaskSet {
```

Comment:
I found it slightly strange that it takes hasUpdates and globalFlags to decide if the function does something or is a no-op (rather than making this check outside), but I see how this slightly simplifies the 3 calls to it and avoids duplicated `if hasUpdates && !globalFlags.NoReRefresh` before every call, so I think it's fine.

Reply:
Maybe this should be called `doPostUpdate()` then.

Reply:
I think the `do` in `doInstall` and `doUpdate` is a bit historical, they don't really do things, the return tasks.  Nowadays they have would have been called `installTasks` etc maybe.  So not a fan of `doPostUpdate`,  would `finishUpdate` sound better?

Reply:
It's green now, so probably not worth to push a patch tweaking the name.